http/okhttp/src/main/java/com/okta/commons/http/okhttp/OkHttpRequestExecutor.java

```diff
@@ -147,6 +170,15 @@ public Response executeRequest(Request request) throws HttpException {
         }
     }
 
+    private String fetchHeaderValueAndRemoveIfExists(Request request, String headerName) {
```

Comment:
Nit: Does `removeHeaderIfPresent(..)` sound like a better method name?

Reply:
the method receives value from headers if the key exists and removes it. IMHO removeHeaderIfPresent does not describe fully what method does.